core/rawdb/database.go

```diff
@@ -35,14 +35,73 @@ import (
 // freezerdb is a database wrapper that enabled freezer data retrievals.
 type freezerdb struct {
 	ethdb.KeyValueStore
-	ethdb.AncientStore
+	chain *freezer
```

Comment:
I'm a bit curious (and lazy:) .. What made this change necessary?

Reply:
Because for ancient store APIs, it requires an additional parameter `typ` and it's used for freezer instance routing. Right now there is only a single instance for chain and later we can add reverse diff and even more.